webmention.php

```diff
@@ -70,6 +70,39 @@ public static function query_var($vars) {
   }
 
   /**
+   * generate a valid expire code. 
+   * Three possible values are valid at any one time, ticks: 0, 1, or 2
+   * 
+   * @return array
+   */
+  public static function expire_code( $tick = 0 ) {
+    $action = 'web mention endpoint';
+    $time_format = 'Y-m-d a';
+    $time_block = 12 * HOUR_IN_SECONDS;
+    $tick = abs( intval( $tick ) );
+    if ( 3 < $tick ) {
+        // something wrong, tick too high/
+        // use default
+        $tick = 0;
+    }
+    $expire_code = date( $time_format, time() - ( $tick * $time_block ) );
+    
+    // always use logged out user code, endpoint may be looked up by a logged in user
+    // while the web mention comes from a logged out user (using curl or similar)
+    $uid = 0;
+    $uid = apply_filters( 'nonce_user_logged_out', $uid, $action );
+    
+    // as above, always use the lgoged out token.
+    $token = '';
+
+
+    $expire_code = wp_hash( $expire_code . '|' . $action . '|' . $uid . '|' . $token, 'nonce' );
```

Comment:
This looks very similar to the code in `wp_create_nonce()`: https://core.trac.wordpress.org/browser/tags/4.1/src/wp-includes/pluggable.php#L1758 But without the substr()? Any reason for that? Should perhaps add a comment on why it's not possible to use `wp_create_nonce()` (I guess the reason is the risk of getting session-dependent data which would stop the endpoint definition from being stateless and "curlable")

Ideally, long term, a patch to WordPress core that would allow a stateless `wp_create_nonce()` to be used out of the box would be great as well I think :) This is a great use case for such a patch.